File: scripts/test_qstash.js
const http = require('http');

const payload = JSON.stringify({
    type: 'test',
    data: 'hello from verification script'
});

const options = {
    hostname: 'localhost',
    port: 3000,
    path: '/webhooks/qstash',
    method: 'POST',
    headers: {
        'Content-Type': 'application/json',
        'Content-Length': payload.length
    }
};

function tryRequest(retries) {
    const req = http.request(options, (res) => {
        console.log(`STATUS: ${res.statusCode}`);
        if (res.statusCode >= 200 && res.statusCode < 300) {
            console.log("SUCCESS");
        }
        res.setEncoding('utf8');
        res.on('data', (chunk) => {
            console.log(`BODY: ${chunk}`);
        });
    });

    req.on('error', (e) => {
        console.error(`problem with request: ${e.message}`);
        if (retries > 0) {
            console.log(`Retrying in 2 seconds... (${retries} left)`);
            setTimeout(() => tryRequest(retries - 1), 2000);
        }
    });

    req.write(payload);
    req.end();
}

tryRequest(5);

File: scripts/test_qstash.py
import urllib.request
import json
import time

url = "http://localhost:3000/webhooks/qstash"
data = {"type": "test", "message": "hello python"}
headers = {
    "Content-Type": "application/json",
    "upstash-signature": "fake_signature_for_testing"
}

encoded_data = json.dumps(data).encode('utf-8')
req = urllib.request.Request(url, data=encoded_data, headers=headers)

max_retries = 5
for i in range(max_retries):
    try:
        with urllib.request.urlopen(req) as response:
            print(f"Status: {response.getcode()}")
            print(response.read().decode('utf-8'))
        break
    except urllib.error.URLError as e:
        print(f"Attempt {i+1} failed: {e}")
        time.sleep(2)
